wallet/src/ui.rs

```diff
@@ -359,13 +368,40 @@ pub enum PrivateKey {
 }
 
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
-pub enum BalanceInfo {
+pub enum Balances {
     /// The balance of a single asset, in a single account.
-    Balance(u64),
+    One(u64),
     /// All the balances of an account, by asset type with asset info.
-    AccountBalances(HashMap<AssetCode, (u64, AssetInfo)>),
+    Account(HashMap<AssetCode, u64>),
     /// All the balances of all accounts owned by the wallet.
-    AllBalances(HashMap<UserAddress, HashMap<AssetCode, (u64, AssetInfo)>>),
+    All {
+        by_account: HashMap<UserAddress, HashMap<AssetCode, u64>>,
+        aggregate: HashMap<AssetCode, u64>,
+    },
+}
+
+impl Balances {
+    pub fn assets(&self) -> Box<dyn Iterator<Item = &AssetCode> + Send + '_> {
+        match self {
+            Self::One(_) => Box::new(empty()),
+            Self::Account(by_asset) => Box::new(by_asset.keys()),
+            Self::All {
+                by_account,
+                aggregate,
+            } => Box::new(
+                by_account
+                    .values()
+                    .flat_map(|by_asset| by_asset.keys())
+                    .chain(aggregate.keys()),
```

Comment:
Isn't each `AssetCode` in `aggregate` guaranteed to be also included in `by_account`? I'm wondering why we need to add both `aggregate.keys()` and `by_asset.keys()`. Can we just use `aggregate` to get the assets?

Reply:
Good point. I used `by_account` since that's the primary source of data, it felt more appropriate to derive everything from that. Got rid of the redundant `.chain(...)`